PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-dbtools/src/main/java/com/microsoft/azure/toolkit/intellij/dbtools/AzureParamsEditorBase.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.intellij.dbtools;
+
+import com.intellij.database.dataSource.DataSourceConfigurable;
+import com.intellij.database.dataSource.url.DataInterchange;
+import com.intellij.database.dataSource.url.FieldSize;
+import com.intellij.database.dataSource.url.template.UrlEditorModel;
+import com.intellij.database.dataSource.url.ui.ParamEditorBase;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.util.NlsContexts;
+import com.intellij.ui.components.JBCheckBox;
+import com.microsoft.azure.toolkit.lib.Azure;
+import com.microsoft.azure.toolkit.lib.auth.AzureAccount;
+import com.microsoft.azure.toolkit.lib.common.model.AzComponent;
+import com.microsoft.azure.toolkit.lib.common.model.AzResource;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTask;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTaskManager;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+public abstract class AzureParamsEditorBase<TCombo extends DbUtilsComboBoxBase<TItem>, TItem extends AzResource> extends ParamEditorBase<TCombo> {
```

Comment:
Maybe its name should have something related to the database? `DbParamEditorBase` or something?